pkg/collector/hotswap_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/nel-collector/pkg/collector"
+)
+
+func TestHotSwapChangesWhichHandlerIsRun(t *testing.T) {
+	codeOkHandler := newDummy(http.StatusOK)
```

Comment:
The factory function isn't doing anything other than constructing an instance, so just create that directly here:

``` go
codeOkHandler := dummyHandler{http.StatusOK}
```

Reply:
Done